game/src/contrib/systems/HealthSystem.java

```diff
@@ -34,22 +34,31 @@ public void execute() {
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
-                .filter(
-                        hsd -> {
-                            if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)
-                                    || hsd.ac
-                                            .getAnimation(AdditionalAnimations.DIE)
-                                            .get()
-                                            .isLooping()) return true;
-                            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
-                                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
-                            }
-                            return hsd.ac.currentAnimation().isFinished();
-                        })
+                .filter(this::checkAnimationStatus)
                 // Remove all dead entities
                 .forEach(this::removeDeadEntities);
     }
 
+    private boolean checkAnimationStatus(HSData hsd) {
+        // if the filtered Entity has no DeathAnimation
+        if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)) {
+            return true;
+        }
+        // if it has a DeathAnimation check if the DeathAnimation is active
+        else {
+            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
+                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
```

Comment:
das finde ich nicht so gut: die methode deutet auf eine prüfung hin (`checkAnimationStatus`), und hier wird definitiv nix abgefragt, sondern ein zustand verändert.

Reply:
die Abfrage wurde in triggerDeathAnimation rausgezogen